### PR TITLE
Lobourre/camera helper update

### PR DESCRIPTION
**Pull request by LPBourret**
Adding property that exposes available compatible formats on the frame source in CameraHelper. Also correcting documentation.

### CODE REVIEW

File: Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraHelper/CameraHelperCode.bind [hunk 1]
```diff
@@ -21,7 +21,4 @@ private void CameraHelper_FrameArrived(object sender, FrameEventArgs e)
 {
   // Gets the current video frame
   VideoFrame currentVideoFrame  = e.VideoFrame;
-
-  // Gets the software bitmap image
-  SoftwareBitmap softwareBitmap = e.SoftwareBitmap;
```
Comment (skommireddi) [inline Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraHelper/CameraHelperCode.bind]: This change is not necessary, I have updated it to get the software bitmap from video frame

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs [hunk 2]
```diff
@@ -125,7 +130,6 @@ public class CameraHelper : IDisposable
             var settings = new MediaCaptureInitializationSettings()
             {
                 SourceGroup = _group,
-                SharingMode = MediaCaptureSharingMode.SharedReadOnly,
```
Comment (skommireddi) [inline Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs:139]: Why is this removed? Is this the default option for MediaCaptureInitializationSettings and doesn't need to be set explicitly?

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs [hunk 3]
```diff
@@ -147,15 +151,34 @@ public class CameraHelper : IDisposable
...
+                // set the format with the higest resolution available by default
+                var defaultFormat = _frameFormatsAvailable.Last();
+                await _frameSource.SetFormatAsync(defaultFormat);
             }
-            catch (UnauthorizedAccessException)
+            catch (UnauthorizedAccessException ex)
             {
                 await Cleanup();
+                Debug.WriteLine(ex.Message);
                 return CameraHelperResult.CameraAccessDenied;
```
Comment (skommireddi) [inline Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs]: Can you remove the Debug lines.

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelperResult.cs [hunk 4]
```diff
@@ -56,6 +56,11 @@ public enum CameraHelperResult
         /// <summary>
         /// Initialization failed due to an exception.
         /// </summary>
-        InitializationFailed_UnknownError
+        InitializationFailed_UnknownError,
+
+        /// <summary>
+        /// Initialization failed, no comprible frame format are exposed via the frame source
```
Comment (skommireddi) [inline Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelperResult.cs]: Spelling wrong

File: docs/controls/CameraPreview.md [hunk 5]
```diff
@@ -25,7 +25,6 @@ The **CameraPreview** control allows to easily preview video in the MediaPlayerE
 private void CameraPreviewControl_FrameArrived(object sender, FrameEventArgs e)
 {
 	var videoFrame = e.VideoFrame;
-	var softwareBitmap = e.SoftwareBitmap;
```
Comment (skommireddi) [inline docs/controls/CameraPreview.md]: Can you revert this as this code has been modified o get software bitmap from video frame

File: docs/helpers/CameraHelper.md [hunk 6]
```diff
@@ -23,20 +23,17 @@ var result = await _cameraHelper.InitializeAndStartCaptureAsync();
...
 }
 
 // Subscribe to get frames as they arrive
 cameraHelper.FrameArrived += CameraHelper_FrameArrived;
 
 private void CameraHelper_FrameArrived(object sender, FrameEventArgs e)
 {
-  // Gets the current video frame
-  VideoFrame currentVideoFrame  = e.VideoFrame;
-
```
Comment (skommireddi) [inline docs/helpers/CameraHelper.md]: Can you revert this change as well, this got modified in the source branch.

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs [hunk 7]
```diff
@@ -34,12 +33,18 @@ public class CameraHelper : IDisposable
...
 
         /// <summary>
         /// Gets the currently selected <see cref="MediaFrameSource"/>MediaFrameSource
         /// </summary>
         public MediaFrameSource FrameSource { get => _frameSource; }
 
+        /// <summary>
+        /// Gets the available MediaFrameFormats on this sources
+        /// </summary>
+        public List<MediaFrameFormat> FrameFormatsAvailable { get => _frameFormatsAvailable; }
```
Comment (skommireddi) [inline Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs]: Would you be adding a setter so that the user can select the format for preview?

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs [hunk 8]
```diff
@@ -34,12 +33,18 @@ public class CameraHelper : IDisposable
...
         private MediaFrameSource _frameSource;
+        private List<MediaFrameFormat> _frameFormatsAvailable;
 
         /// <summary>
         /// Gets the currently selected <see cref="MediaFrameSource"/>MediaFrameSource
         /// </summary>
         public MediaFrameSource FrameSource { get => _frameSource; }
 
+        /// <summary>
+        /// Gets the available MediaFrameFormats on this sources
```
Comment (skommireddi) [inline Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs]: Should say "on the source"

**Review by skommireddi — CHANGES_REQUESTED** [review]
I have made a few changes since we spoke, please update the code from the Camera Helper branch.

File: Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraHelper/CameraHelperCode.bind (hunk 1 above)
Comment (LPBourret) [reply Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraHelper/CameraHelperCode.bind]: by not necessary you mean what? you already changed the .bind?

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs (hunk 2 above)
Comment (LPBourret) [reply Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs:139]: you need to be able to change the format on the frameSource, SharedReadOnly does not allow that to happen

File: Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs (hunk 7 above)
Comment (LPBourret) [reply Microsoft.Toolkit.Uwp/Helpers/CameraHelper/CameraHelper.cs]: no since there is a getter on the FrameSource, I would defer to the API already available on that object

**Review by skommireddi — APPROVED** [review]
